component/core/src/main/java/org/exoplatform/social/core/metadata/attachment/AttachmentServiceImpl.java

```diff
@@ -186,6 +187,43 @@ public ObjectAttachmentList getAttachments(String objectType,
     return new ObjectAttachmentList(attachments, objectType, objectId);
   }
 
+  @Override
+  public void deleteAttachment(String objectType,
+                               String objectId,
+                               String fileId,
+                               Identity userAclIdentity) throws ObjectNotFoundException, IllegalAccessException {
+
+    if (StringUtils.isBlank(objectType)) {
+      throw new IllegalArgumentException("Object type is mandatory");
+    }
+    if (StringUtils.isBlank(objectId)) {
+      throw new IllegalArgumentException("Object id is mandatory");
+    }
+    if (StringUtils.isBlank(fileId)) {
+      throw new IllegalArgumentException("File id is mandatory");
+    }
+    org.exoplatform.social.core.identity.model.Identity userIdentity =
+                                                                     identityManager.getOrCreateUserIdentity(userAclIdentity.getUserId());
+
+    if (userIdentity == null || userIdentity.isDeleted() || !userIdentity.isEnable()) {
+      throw new IllegalStateException("User with id " + userAclIdentity.getUserId() + " isn't valid");
+    }
+
+    if (!hasAccessPermission(userAclIdentity, objectType, objectId)) {
+      throw new IllegalAccessException("User " + userAclIdentity.getUserId()
+          + " doesn't have enough permissions to attach files on object " + objectType + "/" + objectId);
+    }
+    MetadataItem metadataItem =
+                              metadataService.getMetadataItemsByMetadataNameAndTypeAndObject(fileId,
+                                                                                             AttachmentService.METADATA_TYPE.getName(),
+                                                                                             objectType,
+                                                                                             objectId,
+                                                                                             0,
+                                                                                             0)
+                                             .get(0);
+    metadataService.deleteMetadataItem(metadataItem.getId(), true);
```

Comment:
Broadcast attachment deletion event

Reply:
it's already added by the true parameter in this method
https://github.com/Meeds-io/social/blob/feature/meedsv2/component/api/src/main/java/org/exoplatform/social/metadata/MetadataService.java#L185

Reply:
Nope, that's for Metadata, but we need a dedicated broadcast for attachment deletion & update & creation.